packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -123,6 +126,79 @@ void _debugRecordDownStream(String channelTypeName, String name,
   _debugLaunchProfilePlatformChannels();
 }
 
+class BackgroundIsolateBinaryMessenger extends BinaryMessenger {
+  final ReceivePort receivePort = ReceivePort();
+  final Map<int, Completer<ByteData?>> _completers = <int, Completer<ByteData?>>{};
+  int _messageCount = 0;
+
+  static BinaryMessenger? _backgroundIsolateBinaryMessenger;
+
+  static void ensureInitialized(ui.RootIsolateToken token) {
```

Comment:
this will need api docs (you can copy them from my branch)

Reply:
done